Lib/typing.py

```diff
@@ -237,9 +237,12 @@ def _type_repr(obj):
             return obj.__qualname__
         return f'{obj.__module__}.{obj.__qualname__}'
     if obj is ...:
-        return('...')
+        return '...'
     if isinstance(obj, types.FunctionType):
         return obj.__name__
+    if isinstance(obj, tuple):
+        # Special case for `repr` of types with `ParamSpec`:
+        return '[' + ', '.join(_type_repr(t) for t in obj) + ']'
```

Comment:
It _feels_ hacky special-casing `tuple` here. It feels like we should really have a proper internal representation of a _parameters list_ as opposed to a `ParamSpec`, similar to what mypy and pyright have. But that might have performance implications and be messy in all kinds of ways, so probably best left to a future PR :/

Reply:
Yeah, we need to clean things up here. `typing` probably is one of the hardest pure-Python modules in CPython right now. Let's defer this for later.

Reply:
Yeap: https://github.com/python/cpython/pull/98120 :)